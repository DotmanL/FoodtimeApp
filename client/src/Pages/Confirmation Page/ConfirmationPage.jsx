import React, {useEffect} from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { activate } from '../../Redux/User/user.actions';
import Nav  from '../../components/Navbar/Nav';

import { Redirect } from 'react-router-dom';
import {
  Container,
  Details,
}
from './ConfirmationPage.styles'


export const ConfirmationPage = ({activate, match, isAuthenticated}) => {
  
  useEffect(() => {
    activate(match.params.token);
  }, [activate, match.params.token]);
  
 // console.log(match.params.token, 'activate details')

    if(!isAuthenticated) {
      return <Redirect to = "/signin" />
      }

    
  return (
  
   <Container>
       <Nav/>
<Details>
Chill, your account is being verified
</Details>
   </Container>
  )
}


ConfirmationPage.propTypes = {
  activate: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool,
}

const mapStateToProps = state => ({
  isAuthenticated: state.user.isAuthenticated
  })

export default connect(mapStateToProps, {activate}) (ConfirmationPage)
