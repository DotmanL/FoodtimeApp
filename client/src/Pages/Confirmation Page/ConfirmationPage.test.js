import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import { ConfirmationPage } from './ConfirmationPage'

describe('Confirmation component', () => {
  let wrapper;
  let mockActivate;

  beforeEach(() => {
    mockActivate = jest.fn();

    const match = { params: {
      id: 12434,
    },
  }
    const mockProps = {
      activate: mockActivate
    };

    wrapper = shallow(<ConfirmationPage {...mockProps} match={match} />);
  });

  it('should render Confirmation Page component', () => {
    expect(toJson(wrapper))
    .toMatchSnapshot();
  })
  
})