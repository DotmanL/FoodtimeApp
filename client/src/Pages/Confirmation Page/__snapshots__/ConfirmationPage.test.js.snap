// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation component should render Confirmation Page component 1`] = `
<Redirect
  to="/signin"
/>
`;
